### PR TITLE
feat: [ADL/RPL] API for getting extended support period info

### PR DESCRIPTION
**Pull request by lsubash**
HECI interface for getting extended support license period. Efi test is added to call the EPS API for validation.

### CODE REVIEW

**Review by kvparikh89 — COMMENTED** [review]
Please update header in all files.

File: Silicon/AlderlakePkg/Include/Library/HeciMeExtLib.h [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /** @file
   Heci ME Extended Measured boot library.
 
-  Copyright (c) 2022, Intel Corporation. All rights reserved.<BR>
+  Copyright (c) 2022 - 23, Intel Corporation. All rights reserved.<BR>
```
Comment (gdong1) [inline Silicon/AlderlakePkg/Include/Library/HeciMeExtLib.h:4]: Had better merge the 2 commits into 1 commit given one commit just updates the copy right year.

File: Silicon/AlderlakePkg/Include/Library/HeciMeExtLib.h [hunk 2]
```diff
@@ -129,4 +129,19 @@ HeciGetFipsMode (
...
+  @retval EFI_UNSUPPORTED         Current ME mode doesn't support this function
+  @retval EFI_SUCCESS             Command succeeded
+  @retval EFI_DEVICE_ERROR        HECI Device error, command aborts abnormally
+  @retval EFI_TIMEOUT             HECI does not return the buffer before timeout
+  @retval EFI_BUFFER_TOO_SMALL    Message Buffer is too small for the Acknowledge
+**/
+EFI_STATUS
+EFIAPI
+HeciGetEpsState (
+  OUT EPS_GET_STATE_INFO  *GetEpsStateInfo
```
[the remark this answers is not in the document]
Comment (lsubash) [reply Silicon/AlderlakePkg/Include/Library/HeciMeExtLib.h:146]: Done

File: Silicon/AlderlakePkg/Library/HeciMeExtLib/HeciMeExtLib.c [hunk 3]
```diff
@@ -721,3 +721,72 @@ HeciGetFipsMode (
...
+  @retval EFI_UNSUPPORTED         Current ME mode doesn't support this function
+  @retval EFI_SUCCESS             Command succeeded
+  @retval EFI_DEVICE_ERROR        HECI Device error, command aborts abnormally
+  @retval EFI_TIMEOUT             HECI does not return the buffer before timeout
+  @retval EFI_BUFFER_TOO_SMALL    Message Buffer is too small for the Acknowledge
+**/
+EFI_STATUS
+EFIAPI
+HeciGetEpsState (
+  OUT EPS_GET_STATE_INFO  *GetEpsStateInfo
```
[the remark this answers is not in the document]
Comment (lsubash) [reply Silicon/AlderlakePkg/Library/HeciMeExtLib/HeciMeExtLib.c:739]: Done

File: Silicon/AlderlakePkg/Library/HeciMeExtLib/HeciMeExtLib.c [hunk 4]
```diff
@@ -721,3 +721,72 @@ HeciGetFipsMode (
...
+    *GetEpsStateInfo = GetEpsState.Response.eps_info;
+
+    DEBUG ((DEBUG_INFO, "delivery_method    =%d\n", GetEpsStateInfo->delivery_method));
+    DEBUG ((DEBUG_INFO, "license_requested  =%d\n", GetEpsStateInfo->license_requested));
+    DEBUG ((DEBUG_INFO, "license_installed  =%d\n", GetEpsStateInfo->license_installed));
+    DEBUG ((DEBUG_INFO, "license_permits    =%d\n", GetEpsStateInfo->license_permits));
+
+  } else {
+    DEBUG ((DEBUG_INFO, "Error in EPS Info population \n"));
+    return EFI_DEVICE_ERROR;
```
[the remark this answers is not in the document]
Comment (lsubash) [reply Silicon/AlderlakePkg/Library/HeciMeExtLib/HeciMeExtLib.c]: Done

**Comment by kvparikh89** [comment]
@lsubash: Can you please work on closing this PR?

File: Silicon/AlderlakePkg/Include/Library/HeciMeExtLib.h (hunk 1 above)
Comment (lsubash) [reply Silicon/AlderlakePkg/Include/Library/HeciMeExtLib.h]: done

**Review by kvparikh89 — APPROVED** [review]
LGTM

**Review by gdong1 — APPROVED** [review]

**Pull request merged**